src/middlewared/middlewared/job.py

```diff
@@ -628,6 +633,7 @@ def __encode__(self, raw_result=True):
             'logs_excerpt': self.logs_excerpt,
             'progress': self.progress,
             'result': result,
+            **({'result_encoding_error': result_encoding_error} if result_encoding_error is not None else {}),
```

Comment:
This line is strange and I'm having problems groking it. Does this mean the `result_encoding_error` key only gets inserted into the object if `result_encoding_error` object is not None? Does it make sense to just do something like
```
'result': result,
'result_encoding_error': result_encoding_error,
...
```

Reply:
Also curious why we're crashing?

Reply:
If the job result schema does not match the actual result, we'll fail to remove secrets and crash